spec/api/user_api/v1/documents_spec.rb

```diff
@@ -19,11 +19,27 @@
       }
     end
 
+    let!(:optional_params) do
+      {
+        metadata: {
+          country: Faker::Address.country
+        }
+      }
+    end
+
     it 'Checks if params are ok and returns success' do
       post '/api/v1/documents', headers: headers, params: params
       expect(response.status).to eq(201)
     end
 
+    it 'Creates document with optional params and returns success' do
```

Comment:
Please, add a new `describe` block under your `it` block
Add into it:
`let(:do_request) { post '/api/v1/documents', headers: headers, params: params.merge(optional_params) }`
and use inside your `it` blocks like expect { do_request }.to change { account.reload.documents.count }